### PR TITLE
Improve trivia preservation when converting methods into a property.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/21460

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
VB side coming next.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs [hunk 1]
```diff
@@ -220,6 +324,42 @@ int Goo
...
+    }
+
+    int [||]GetGoo()
+    {
+    }
+#endif
+}",
+@"class C
+{
+
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs]: 😕 I can't figure out why *only* this test ends up with the extra trivia.

Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs]: It's grabbing it off of the getter.  The assumption being that the getter is normally before the setter.  Here it's suboptimal, but not terrible.  We can consider trying to do better in the cases where the getter is after the setter.  But i just wanted to ensure we weren't totally breaking code by removing preprocessor directives as the highest order thing to fix.

File: src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs [hunk 2]
```diff
@@ -64,11 +60,14 @@ public void RemoveSetMethod(SyntaxEditor editor, SyntaxNode setMethodDeclaration
...
-                ConvertMethodsToProperty(
-                    documentOptions, parseOptions,
-                    semanticModel, editor.Generator,
-                    getAndSetMethods, propertyName, nameChanged));
+            var newProperty = ConvertMethodsToProperty(
+                documentOptions, parseOptions,
+                semanticModel, editor.Generator,
+                getAndSetMethods, propertyName, nameChanged);
+
+            // newProperty = newProperty.WithTriviaFrom(getMethodDeclaration);
```
Comment (sharwell) [inline src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs]: :bulb: dead code?

Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/ReplaceMethodWithProperty/CSharpReplaceMethodWithPropertyService.cs]: Indeed.

**Pull request merged**